amgut/test/test_auth_handlers.py

```diff
@@ -0,0 +1,133 @@
+    def test_get(self):
+        response = self.get('/auth/register/')
+        self.assertEqual(response.code, 200)
+        port = self.get_http_port()
+        self.assertEqual(response.effective_url,
+                         'http://localhost:%d/auth/register/' % port)
+
+    def test_post(self):
+        # test reaction to invalid kit IDs
+        response = self.post('/auth/register/',
+                             {'kit_id': 'wrong_kit_id',
+                              'password': 'test',
+                              'email': 'e@mail',
+                              'email2': 'e@mail',
+                              'participantname': 'joe',
+                              'address': 'street',
+                              'city': 'metropolis',
+                              'state': 'XX',
+                              'zip': '12345',
+                              'country': 'Afghanistan'})
+        port = self.get_http_port()
+        self.assertEqual(
+            response.effective_url,
+            'http://localhost:%d/?loginerror=%s' %
+            (port, url_escape(text_locale['handlers']['INVALID_KITID'])))
+
+        unregistered_kits = ag_data.get_all_handout_kits()
+        if len(unregistered_kits) <= 0:
+            raise Exception(("Test cannot be performed. No more un-registered "
+                             "kit IDs in the database. Re-set test database!"))
+        else:
+            response = self.post('/auth/register/',
+                                 {'kit_id': unregistered_kits[0],
+                                  'password': 'test',
+                                  'email': 'e@mail',
+                                  'email2': 'e@mail',
+                                  'participantname': 'joe',
+                                  'address': 'street',
+                                  'city': 'metropolis',
+                                  'state': 'XX',
+                                  'zip': '12345',
+                                  'country': 'Afghanistan'})
+            port = self.get_http_port()
+            self.assertEqual(response.code, 200)
+            self.assertEqual(
+                response.effective_url,
+                'http://localhost:%d/?next=%s' %
+                (port, url_escape('/authed/portal/')))
+
+            # Check that registration changed the DB. A proxy to test that no
+            # ValueError is raised.
+            self.assertNotEqual(ag_data.get_user_for_kit(unregistered_kits[0]),
+                                'something')
+
+
+class TestAuthLoginHandler(TestHandlerBase):
+    def test_get(self):
+        response = self.get('/auth/login/')
+        self.assertEqual(response.code, 200)
+        # make sure redirect happened properly
+        port = self.get_http_port()
+        self.assertEqual(response.effective_url, 'http://localhost:%d/' % port)
+
+        # check if url is wrong. SJ: Is this in general a good idea?
```

Comment:
I don't think this test is needed, because is not actually testing the Handler that we wrote, but rather the entire "webserver.py" and a basic functionality of tornado. Safe to remove :-)